praw/models/reddit/redditor.py

```diff
@@ -178,6 +178,20 @@ def _friend(self, method, data):
         url = API_PATH["friend_v1"].format(user=self)
         self._reddit.request(method, url, data=dumps(data))
 
+    def available(self) -> bool:
```

Comment:
I wonder if this should be a class method, rather than an instance method. It seems a little strange to make a `Redditor` instance when none of the normal methods on a Redditor work on said instance (unless it happened to be registered).

Thinking as I type, the usage in this way would probably need to be:

    praw.models.Redditor.username_available(...)

That approach feels a little bit verbose. An alternative could be on the `User` object:

    reddit.user.name_available(...)

That approach seems a little weird because `user` pertains to the current authentication. Finally, while I don't like adding a ton of stuff in the top-level namespace, this is very much a one-off method so it could be referenced like:

    reddit.username_available(...)

Reply:
This endpoint isn't the only one that doesn't really fit in with the others. For example, api/v1/scopes is listed in an example in docs/tutorials/refresh_token.rst, but PRAW doesn't have a method for it. I know that the captcha endpoints were taken out of here a while ago, but api/requires_captcha (which is mislabeled in the "documentation" as api/needs_captcha) is like this too. You could make a miscellaneous class for orphaned methods like these.

Reply:
Personally, I feel this would fit best be in the top-level namespace.

Reply:
@MaybeNetwork can you move the function `username_available` into the top-level?

Reply:
I can do that. I do have a question though. There are a bunch of other oauth endpoints that can be used without creating a class. For example, api/fetch_title is an unlisted oauth endpoint used by the main site to fetch the default name for a link submission. It can be used like this:

    reddit=praw.Reddit()
    reddit.request("POST", "api/fetch_title", data={"url": "https://docs.python.org/3/library"})

which yields this response:

    {'json': {'errors': [], 'data': {'title': 'The Python Standard Library — Python 3.9.5 documentation'}}}

Other examples include:

- Scopes: `reddit.request("GET", "api/v1/scopes")`
- Sitewide rules: `reddit.request("GET", "/api/sitewide_rules")`
- Livestream rules: `reddit.request("GET", "/api/livestream_rules")`
- Generate username: `reddit.request("GET", "api/v1/generate_username")` (my favorite)
- Crosspostable subreddits: `reddit.request("GET", "/api/crosspostable_subreddits")` (maybe not the best example)
- Purchase package details: `reddit.request("GET", "api/v2/gold/purchase_package_details")`

and more. Where would these go?

Reply:
I've moved username_available up

Reply:
Good question about those other methods. They may not be ones that we expose unless someone has a good reason to include them -- we don't want to add methods just because there are endpoints available for them.